user-guide/Advanced_Modules/Dashboards_and_Low_Code_Apps/GQI/Extensions/Ad_hoc_data/Ad_hoc_Tutorials_GQIDMS.md

```diff
@@ -0,0 +1,121 @@
+---
+uid: Ad_hoc_Tutorials_GQIDMS
+---
+
+# Building a GQI data source that retrieves data from a DMS
+
+In this tutorial, you will learn how you can create a GQI data source that retrieves the active client connections from your DMS.
+
+Expected duration: 15 minutes.
+
+> [!NOTE]
+> This tutorial uses DataMiner version 10.3.4/10.4.0 onwards.
+
+## Prerequisites
+
+- [Visual Studio](https://visualstudio.microsoft.com/downloads/) with [DataMiner Integration Studio](xref:Installing_and_configuring_DataMiner_Integration_Studio)
+
+## Overview
+
+- [Step 1: Create the ad hoc data source](#step-1-create-the-ad-hoc-data-source)
+- [Step 2: Fetch the client connections from the DMS](#step-2-fetch-the-client-connections-from-the-dms)
+- [Step 3: Parse the responses into a GQIPage](#step-3-parse-the-responses-into-a-gqipage)
+
+## Step 1: Create the ad hoc data source
+
+1. Create a new class that implements the [IGQIDatasource](xref:GQI_IGQIDataSource) interface.
+
+> [!NOTE]
```

Comment:
I think this note isn't nested correctly. In the preview, the numbered list now goes 1, 1, 2